protocols/perf/examples/perf-client.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Protocol Labs.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use clap::Parser;
+use futures::{executor::block_on, future::Either, StreamExt};
+use libp2p_core::{
+    identity, muxing::StreamMuxerBox, transport::OrTransport, upgrade, Multiaddr, PeerId, Transport,
+};
+use libp2p_dns::DnsConfig;
+use libp2p_perf::RunParams;
+use libp2p_swarm::{SwarmBuilder, SwarmEvent};
+use log::info;
+
+#[derive(Debug, Parser)]
+#[clap(name = "libp2p perf client")]
+struct Opts {
+    #[arg(long)]
+    server_address: Multiaddr,
+}
+
+fn main() {
```

Comment:
Why are these examples and not proper crate binaries? I think the fact that we ship them in a Docker container makes them binary worth instead of just "examples".

Reply:
My flawed thought process, moving everything to `examples/` after starting with `src/bin/`, was the hope to prevent circular dependencies in `Cargo.toml/[dependencies]`.

Thinking about it some more, as long as I don't depend on `libp2p` directly, I should be fine.

Reply:
> My flawed thought process, moving everything to `examples/` after starting with `src/bin/`, was the hope to prevent circular dependencies in `Cargo.toml/[dependencies]`.
> 
> Thinking about it some more, as long as I don't depend on `libp2p` directly, I should be fine.

The other option would be to create an entirely separate crate, like `interop-tests` instead of building binaries within `libp2p-perf`.

I don't mind either way. With latter, you can easily depend on `libp2p`. With the former, you need to depend on select crates that you want to use.